Source/Concurrency/YieldTypeChecker.cs

```diff
@@ -42,43 +48,39 @@ private static int MoverTypeToLabel(MoverType moverType)
                     throw new InvalidEnumArgumentException();
             }
         }
-
-        static List<Tuple<int, int, int>> ASpec = new List<Tuple<int, int, int>>
-        { // initial: RM, final: LM
-            new Tuple<int, int, int>(RM, Y, LM),
-            new Tuple<int, int, int>(LM, Y, LM),
-            new Tuple<int, int, int>(LM, B, LM),
-            new Tuple<int, int, int>(LM, R, LM),
-            new Tuple<int, int, int>(LM, L, LM),
-            new Tuple<int, int, int>(LM, A, LM),
-            new Tuple<int, int, int>(RM, P, RM),
-            new Tuple<int, int, int>(LM, P, LM)
-        };
-        static List<Tuple<int, int, int>> BSpec = new List<Tuple<int, int, int>>
-        { // initial: LM, final: RM
-            new Tuple<int, int, int>(LM, Y, RM),
-            new Tuple<int, int, int>(LM, Y, LM),
-            new Tuple<int, int, int>(LM, B, LM),
-            new Tuple<int, int, int>(LM, R, LM),
-            new Tuple<int, int, int>(LM, L, LM),
-            new Tuple<int, int, int>(LM, A, LM),
-            new Tuple<int, int, int>(RM, P, RM),
-            new Tuple<int, int, int>(LM, P, LM)
+        
+        static List<Tuple<int, int, int>> ParenthesisSpec = new List<Tuple<int, int, int>>
+        { // initial: Before, final: After
+            new Tuple<int, int, int>(BEFORE, P, BEFORE),
+            new Tuple<int, int, int>(BEFORE, Y, INSIDE),
+            new Tuple<int, int, int>(BEFORE, Y, AFTER),
+            new Tuple<int, int, int>(INSIDE, Y, INSIDE),
+            new Tuple<int, int, int>(INSIDE, B, INSIDE),
+            new Tuple<int, int, int>(INSIDE, R, INSIDE),
+            new Tuple<int, int, int>(INSIDE, L, INSIDE),
+            new Tuple<int, int, int>(INSIDE, A, INSIDE),
+            new Tuple<int, int, int>(INSIDE, P, INSIDE),
+            new Tuple<int, int, int>(INSIDE, I, INSIDE),
+            new Tuple<int, int, int>(INSIDE, Y, AFTER),
+            new Tuple<int, int, int>(AFTER, P, AFTER),
         };
-        static List<Tuple<int, int, int>> CSpec = new List<Tuple<int, int, int>>
+        
+        static List<Tuple<int, int, int>> AtomicitySpec = new List<Tuple<int, int, int>>
         { // initial: {RM, LM}, final: {RM, LM}
+            new Tuple<int, int, int>(RM, I, RM),
             new Tuple<int, int, int>(RM, B, RM),
             new Tuple<int, int, int>(RM, R, RM),
             new Tuple<int, int, int>(RM, Y, RM),
             new Tuple<int, int, int>(RM, B, LM),
             new Tuple<int, int, int>(RM, R, LM),
             new Tuple<int, int, int>(RM, L, LM),
             new Tuple<int, int, int>(RM, A, LM),
+            new Tuple<int, int, int>(LM, I, LM),
             new Tuple<int, int, int>(LM, B, LM),
             new Tuple<int, int, int>(LM, L, LM),
             new Tuple<int, int, int>(LM, Y, RM),
             new Tuple<int, int, int>(RM, P, RM),
-            new Tuple<int, int, int>(LM, P, LM)
+            new Tuple<int, int, int>(LM, P, LM),
```

Comment:
While making this change, I realized that the following two edges are redundant:

new Tuple<int, int, int>(RM, B, LM),            
new Tuple<int, int, int>(RM, R, LM),

Do you agree?

Reply:
Yes, I think that's correct.

The computed simulation relation will be smaller (i.e., there are cases where previously the initial state could be mapped to {RM, LM}, while now it will be just {LM}), but the existence of a simulation relation will be unchanged.